pkg/apis/core/validation/validation.go

```diff
@@ -5175,7 +5179,7 @@ func ValidateReplicationControllerSpec(spec *core.ReplicationControllerSpec, fld
 	allErrs = append(allErrs, ValidateNonnegativeField(int64(spec.MinReadySeconds), fldPath.Child("minReadySeconds"))...)
 	allErrs = append(allErrs, ValidateNonEmptySelector(spec.Selector, fldPath.Child("selector"))...)
 	allErrs = append(allErrs, ValidateNonnegativeField(int64(spec.Replicas), fldPath.Child("replicas"))...)
-	allErrs = append(allErrs, ValidatePodTemplateSpecForRC(spec.Template, spec.Selector, spec.Replicas, fldPath.Child("template"), opts)...)
+	allErrs = append(allErrs, ValidatePodTemplateSpecForRC(spec.Template, nil, spec.Selector, spec.Replicas, fldPath.Child("template"), opts)...)
```

Comment:
This function is called from `ValidateReplicationController` and `ValidateReplicationControllerUpdate`, so you need to also update those to pass oldSpec down